nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -1125,6 +1130,10 @@ def decode(
         bos_id=None,  # If bos=None, will use tokenizer.bos_id unless explicitly set to something else.
         sample_token_fn=sample_token_greedy,
         predicted_tokens_dec=None,
+        beam_size: Optional[int] = None,
```

Comment:
Let's change it to:
`sampling_method="beam-search"/"greedy"/"topkp"`
and
`sampling_kwargs={...}`
and adapt nemo and bionemo to this format. I will work with you on that (need permissions to be able to work on it)